js/main.js

```diff
@@ -1,18 +1,79 @@
-function checkRange (min, max) {
-  if (min < 0 || min >= max) {
-    throw new Error('Недопустимый диапазон');
-  }
+function getRandomPositiveInteger (min, max) {
+  const lower = Math.ceil(Math.min(Math.abs(min), Math.abs(max)));
+  const upper = Math.floor(Math.max(Math.abs(min), Math.abs(max)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+
+  return Math.floor(result);
 }
 
-function getRandomInteger (min, max) {
-  checkRange(min, max);
-  return Math.floor(Math.random() * (max - min + 1) + min);
+function getRandomPositiveFloat (min, max, digits = 1) {
+  const lower = Math.min(Math.abs(min), Math.abs(max));
+  const upper = Math.max(Math.abs(min), Math.abs(max));
+  const result = Math.random() * (upper - lower) + lower;
+
+  return Number(result.toFixed(digits));
 }
 
-function getFloatingNumber (min, max, decimalPoint) {
-  checkRange(min, max);
-  return Number((Math.random() * (max - min) + min).toFixed(decimalPoint));
+const AUTHOR = {
+  avatar: `img/avatars/user${  getRandomPositiveInteger(1, 10)  }.png`,
+};
+
+const locationA = {
+  lat: getRandomPositiveFloat(35.65, 35.7, 5),
+  lng: getRandomPositiveFloat(139.7, 139.8, 5),
+};
+
+const getRandomArrayElement = (elements) => elements[getRandomPositiveInteger(0, elements.length - 1)];
+
+const CHECKIN = ['12:00', '13:00', '14:00'];
+const CHECKOUT = ['12:00', '13:00', '14:00'];
+const TYPE = ['palace', 'flat', 'house', 'bungalow', 'hotel'];
+const FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+const PHOTOS = ['https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg', 'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg', 'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg'];
+
+// https://qastack.ru/programming/2450954/how-to-randomize-shuffle-a-javascript-array Функция взята отсюда
+
+function shuffle(array) {
+  let currentIndex = array.length, temporaryValue, randomIndex;
+
+  // While there remain elements to shuffle...
+  while (0 !== currentIndex) {
+
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = array[currentIndex];
+    array[currentIndex] = array[randomIndex];
+    array[randomIndex] = temporaryValue;
```

Comment:
Переменная temporaryValue не нужна, а эту запись можно заменить на запись с использованием деструктуризации
```suggestion
    [array[currentIndex], array[randomIndex]] = [array[randomIndex], array[currentIndex]];
```